### PR TITLE
w5e4 by Liron Farzam

### PR DESCRIPTION
**Pull request by lironfarzam**

### CODE REVIEW

**Comment by talbarfm** [comment]
Give more detailed commit message

**Comment by talbarfm** [comment]
Pick a different file name -  the w5e3 is a bad name - its not understandable to anyone but you

File: w5e4.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# "w5e4"
+
+# connected vessels
+'''
```
Comment (talbarfm) [inline w5e4.py]: """ not ''

File: w5e4.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ Write a function called interleave that gets one or more iterable parameters, and returns a list of intertwined members.
+'''
+
+# function to interleave lists
+def interleave(*iterable):
```
Comment (talbarfm) [inline w5e4.py:3]: write the documents inside the function

Comment (talbarfm) [inline w5e4.py:3]: iterables* ?

File: w5e4.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
+    #Run like this on the loop because it is important that the first members of each list appear first
```
Comment (talbarfm) [inline w5e4.py]: space after #

File: w5e4.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+# connected vessels
+'''
+ Write a function called interleave that gets one or more iterable parameters, and returns a list of intertwined members.
+'''
+
+# function to interleave lists
+def interleave(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
```
Comment (talbarfm) [inline w5e4.py]: dont use single letter variables

File: w5e4.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+# connected vessels
+'''
+ Write a function called interleave that gets one or more iterable parameters, and returns a list of intertwined members.
+'''
+
+# function to interleave lists
+def interleave(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
```
Comment (talbarfm) [inline w5e4.py:11]: result? pick a meaningful name

Comment (talbarfm) [reply w5e4.py:11]: I want to enjoy reading the code, its supposed to be as close to English a possible

File: w5e4.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            if j < len(iterable[i]):
+                yield (iterable[i][j])
+        j += 1
+
+
+# our_generator = interleave_generator('abc', [1, 2, 3], ('!', '@', '#', '$'))
+# for i in our_generator:
+#     print(i)
+
+
```
Comment (talbarfm) [inline w5e4.py:43]: Don't use imports in the middle of the file - separate to different files

File: w5e4.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+'''
+
+# function to interleave lists
+def interleave(*iterable):
+    longest = max(len(i) for i in iterable)
+    result = []
+    j = 0
+    #Run like this on the loop because it is important that the first members of each list appear first
+    while j < longest:
+        for i in range(len(iterable)):
```
Comment (talbarfm) [inline w5e4.py]: try to reduce these for loops using one of these tools - map\filter\reduce\listcomprehention\itertools.chain\zip

File: w5e4.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            if j < len(iterable[i]):
+                result.append(iterable[i][j])
+        j += 1
+
+    return result
+
+# print(interleave('abc', [1, 2, 3], ('!', '@', '#', '$')))
```
Comment (talbarfm) [inline w5e4.py:24]: ???

File: w5e4.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    result = []
+    j = 0
+    while j < longest:
+        for i in range(len(iterable)):
+            if j < len(iterable[i]):
+                yield (iterable[i][j])
+        j += 1
+
+
+# our_generator = interleave_generator('abc', [1, 2, 3], ('!', '@', '#', '$'))
```
Comment (talbarfm) [inline w5e4.py:39]: Why do you have these comments inside the file?

File: w5e4.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+# for i in our_generator:
+#     print(i)
+
+
+#Harry is rational but not terrible
+"""
+The librarian who was on his way to bring me the coveted copy of "Harry Potter and the Rational Method" stumbled.
```
Comment (talbarfm) [inline w5e4.py:46]: Im really hoping you didnt translate all of that on your own :) You dont have to do that

File: w5e4.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+During the exercise you may need to combine work with some libraries, including those we have not studied.
+"""
+import os
+import re
+import codecs
+from bs4 import BeautifulSoup
+
+#function get num in string ang make it 3 digits long and return it as a string
+def three_digits(num):
+    if len(num) == 3:
```
Comment (talbarfm) [inline w5e4.py]: https://stackoverflow.com/questions/3505831/in-python-how-do-i-convert-a-single-digit-number-into-a-double-digits-string - try to google search it

File: w5e4.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        file = codecs.open(path + "/" + filename, "r", encoding="utf8")
+        html = file.read()
+        soup = BeautifulSoup(html, 'html.parser')
+        div = soup.find(id="chapter-title")
+        chapterNum, chapterTitel = getCapterNumAndTitel(div.get_text())
+        file.close()
+        new_name = chapterNum + chapterTitel
+        os.rename(os.path.join(path + "/", filename), os.path.join(path + "/", new_name + '.html'))
+
+rename("Harry")
```
Comment (talbarfm) [inline w5e4.py]: Add a newline at eof

File: w5e4.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+        if temp[i] != "\n" and temp[i] != ":":
+            ret2 += temp[i]
+        i = i+1
+
+    return three_digits(ret1.replace(" ", "")),ret2
+
+
+def rename(path):
+
+    str=""
```
Comment (talbarfm) [inline w5e4.py]: spaces around all operators

File: w5e4.py [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    while i < len(temp) and temp[i] != "<":
+        if temp[i] != "\n" and temp[i] != ":":
+            ret2 += temp[i]
+        i = i+1
+
+    return three_digits(ret1.replace(" ", "")),ret2
+
+
+def rename(path):
+
```
Comment (talbarfm) [inline w5e4.py]: Remove this newline

File: w5e4.py [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+    if len(num) == 3:
+        return num
+    elif len(num) == 2:
+        return '0' + num
+    elif len(num) == 1:
+        return '00' + num
+    else:
+        return '000'
+
+# function get string cut the head and split to num and title
```
Comment (talbarfm) [inline w5e4.py]: Your functions should look like this -
Capture

File: w5e4.py [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+def getCapterNumAndTitel(str):
+    head = "Chapter"
+    temp = str[len(head):]
+    ret1 =""
+    ret2 = ""
+    i = 0
+
+    #Capter num
+    while temp[i] != ":":
```
Comment (talbarfm) [inline w5e4.py:83]: I would exclude each section to a different function - if you can define it as a section, you might as well give it a matching name

File: w5e4.py [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+    else:
+        return '000'
+
+# function get string cut the head and split to num and title
+def getCapterNumAndTitel(str):
+    head = "Chapter"
+    temp = str[len(head):]
+    ret1 =""
+    ret2 = ""
+    i = 0
```
Comment (talbarfm) [inline w5e4.py:80]: All of these variables should have better names

File: w5e4.py [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+        return num
+    elif len(num) == 2:
+        return '0' + num
+    elif len(num) == 1:
+        return '00' + num
+    else:
+        return '000'
+
+# function get string cut the head and split to num and title
+def getCapterNumAndTitel(str):
```
Comment (talbarfm) [inline w5e4.py]: snake_case not camelCase

File: w5e4.py [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+        i = i+1
+
+    return three_digits(ret1.replace(" ", "")),ret2
+
+
+def rename(path):
+
+    str=""
+    for filename in os.listdir(path):
+        file = codecs.open(path + "/" + filename, "r", encoding="utf8")
```
Comment (talbarfm) [inline w5e4.py]: Change the names to better names in here as well

**Review by talbarfm — CHANGES_REQUESTED** [review]
Pay attention to read about existing implantations and libraries online before you go and implement something on your own, also pay attention to conventions - pep 8, documentation and variable names.

File: w5e4.py (hunk 1 above)
Comment (lironfarzam) [reply w5e4.py]: """

**Pull request merged**